src/main/java/ox/app/states/PreEndState.java

```diff
@@ -0,0 +1,107 @@
+package ox.app.states;
+
+import ox.app.game.ScoreBoard;
+import ox.app.game.Board;
+import ox.app.game.Player;
+import ox.app.utility.RoundBuffer;
+
+import java.util.function.Consumer;
+
+public class PreEndState implements GameState {
+
+    private final Board board;
+    private final RoundBuffer playerBuffer;
+    private final Player winningPlayer;
+    private final Consumer<String> output;
+    private final ScoreBoard scoreBoard;
+    private final int currentBoardSize;
+    private final int winningStroke;
+
+    private static final int POINTS_FOR_WIN = 3;
+    private static final int POINTS_FOR_DRAW = 1;
+    private static final int POINTS_FOR_LOST = 0;
+    private static final int NUMBER_OF_ROUNDS = 3;
+
+    private final String winnerMessage;
+    private static final String DRAW_MESSAGE = "It's a draw!";
+
+    private int roundCounter;
+
+    public PreEndState(RoundBuffer playerBuffer, Consumer<String> output, Board board, ScoreBoard scoreBoard, int roundCounter, int currentBoardSize,int winningStroke) {
+        this.playerBuffer = playerBuffer;
+        this.winningPlayer = playerBuffer.takePlayer();
+        winnerMessage = winningPlayer + " wins the round!";
+        this.output = output;
+        this.board = board;
+        this.scoreBoard = scoreBoard;
+        this.roundCounter = roundCounter;
+        this.currentBoardSize = currentBoardSize;
+        this.winningStroke = winningStroke;
+    }
+
+    @Override
+    public void showState() {
+        pointsDisposer();
+    }
+
+    @Override
+    public GameState nextState(String input) {
+        if (roundCounter > NUMBER_OF_ROUNDS) {
+            return new SummaryState(scoreBoard);
+        }
+        else {
+            return new RunState(playerBuffer,output,createNewBoard(),scoreBoard,roundCounter,winningStroke);
+        }
+    }
+
+    private Board createNewBoard() {
+        int height = board.getHeight();
+        int width = board.getWidth();
+        return Board.newBoard(width,height);
+    }
+
+    private void pointsDisposer() {
+        if (currentBoardSize>0) {
+            winningPoints();
+            printRoundMessage(winnerMessage);
+        } else {
+            drawPoints();
+            printRoundMessage(DRAW_MESSAGE);
+        }
+        roundCounter++;
+    }
+
+    private void winningPoints() {
+        scoreBoard.addPoint(winningPlayer, POINTS_FOR_WIN);
+        playerBuffer.swapPlayers();
+        scoreBoard.addPoint(playerBuffer.takePlayer(),POINTS_FOR_LOST);
```

Comment:
Is it needed to call this function if POINTS_FOR_LOST is equal to 0?

Do you need this constant?

Reply:
It is needed, because in that case map which holds Integer would show null except of zero. Constant is to prevent magical numbers, and in further code changes, helps to change it only in one place. Also helps in readability.